packages/api/lambdas/publish-reports.js

```diff
@@ -227,16 +234,42 @@ async function publishReportSnsMessages(eventMessage, isTerminalStatus, isFailed
  * @returns {Object} - Execution step message or execution input message
  */
 async function getFailedExecutionMessage(inputMessage) {
-  const executionArn = getMessageExecutionArn(inputMessage);
+  try {
+    const executionArn = getMessageExecutionArn(inputMessage);
+    const { events } = await StepFunctions.getExecutionHistory({ executionArn });
+
+    const stepFailedEvents = events.filter(
+      (event) =>
+        ['ActivityFailed', 'LambdaFunctionFailed'].includes(event.type)
+    );
+    const lastStepFailedEvent = stepFailedEvents[stepFailedEvents.length - 1];
+    const failedStepExitedEvent = getStepExitedEvent(events, lastStepFailedEvent);
```

Comment:
I made this a reusable function because getting the "TaskStateExited" event for any lambda/activity completion/failure step is a generic operation which seems possibly useful elsewhere, but I'm not attached to it by any means